### PR TITLE
Wishlist controller

### PR DESCRIPTION
**Pull request by pablopolyte**

### CODE REVIEW

File: classes/WishList.php [hunk 1]
```diff
@@ -317,114 +317,275 @@ public static function getProductByIdCustomer($id_wishlist, $id_customer, $id_la
...
+        }
+
+        return \Db::getInstance(_PS_USE_SQL_SLAVE_)->getRow('
+        SELECT w.`id_wishlist`, w.`name`, w.`id_customer`, c.`firstname`, c.`lastname`
+        FROM `' . _DB_PREFIX_ . 'wishlist` w
+        INNER JOIN `' . _DB_PREFIX_ . 'customer` c ON c.`id_customer` = w.`id_customer`
+        WHERE `token` = \'' . pSQL($token) . '\'');
+    }
+
+    public static function refreshWishList($id_wishlist)
```
Comment (matks) [inline classes/WishList.php:462]: Waho, this function is so huge and do things you should be afraid of 😅 (deleting data)

Maybe make it easier to read (and consequently easier to maintain) by using private functions ? 😉 
```
ublic static function refreshWishList($id_wishlist)
    {
        $old_carts = $this->getOldCarts();

        if (isset($old_carts) and $old_carts != false) {
            $this->deleteOldCarts($old_carts);
        }

        $freshWishlistContent = $this->getFreshWishes();
        $currentWishlistContent = $this->getCurrentWishlist();

        if (isset($currentWishlistContent) and $currentWishlistContent != false) {
            $this->update...
        }
```

File: classes/WishList.php [hunk 2]
```diff
@@ -317,114 +317,275 @@ public static function getProductByIdCustomer($id_wishlist, $id_customer, $id_la
...
+     * Increment counter
+     *
+     * @return bool succeed
+     */
+    public static function incCounter($id_wishlist)
+    {
+        if (!\Validate::isUnsignedId($id_wishlist)) {
+            die(\Tools::displayError());
+        }
+        $result = \Db::getInstance()->getRow('
```
Comment (matks) [inline classes/WishList.php:544]:
```suggestion
        $result = $this->getWishlistCounter(); 
```

File: controllers/front/view.php [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        $this->context = Context::getContext();
+    }
+
+    public function initContent()
+    {
+        parent::initContent();
+        $token = Tools::getValue('token');
+        $module = new BlockWishList();
+
+        if (false === empty($token)) {
```
Comment (matks) [inline controllers/front/view.php:37]: You could use a [Return early]() here to remove 1 indentation level:
```
if (false === empty($token)) {
    $this->setTemplate('module:blockwishlist/views/templates/front/view.tpl');
    return;
}
```

File: controllers/front/view.php [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+            );
+
+            $nb_products = count($products);
+            $priority_names = [0 => $module->l('High'), 1 => $module->l('Medium'), 2 => $module->l('Low')];
+
+            for ($i = 0; $i < $nb_products; ++$i) {
+                $obj = new Product((int) $products[$i]['id_product'], true, $this->context->language->id);
+                if (!Validate::isLoadedObject($obj)) {
+                    continue;
+                } else {
```
Comment (matks) [inline controllers/front/view.php:56]: Useless `else` because you use `continue` in the if 😉

File: controllers/front/view.php [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+                $this->context->language->id,
+                null,
+                true
+            );
+
+            $nb_products = count($products);
+            $priority_names = [0 => $module->l('High'), 1 => $module->l('Medium'), 2 => $module->l('Low')];
+
+            for ($i = 0; $i < $nb_products; ++$i) {
+                $obj = new Product((int) $products[$i]['id_product'], true, $this->context->language->id);
```
Comment (matks) [inline controllers/front/view.php:53]:
```suggestion
                $product = new Product((int) $products[$i]['id_product'], true, $this->context->language->id);
```
Use meaningful variable names, as they help you write readable code

File: controllers/front/view.php [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+                if (!Validate::isLoadedObject($obj)) {
+                    continue;
+                } else {
+                    $products[$i]['priority_name'] = $priority_names[$products[$i]['priority']];
+                    $quantity = Product::getQuantity((int) $products[$i]['id_product'], $products[$i]['id_product_attribute']);
+                    $products[$i]['attribute_quantity'] = $quantity;
```
Comment (matks) [inline controllers/front/view.php:59]: Some empty lines between statements would make it easier to read ❤️

File: controllers/front/view.php [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+                    $products[$i]['product_quantity'] = $quantity;
+                    $products[$i]['allow_oosp'] = $obj->isAvailableWhenOutOfStock((int) $obj->out_of_stock);
+                    if ($products[$i]['id_product_attribute'] != 0) {
```
Comment (matks) [inline controllers/front/view.php:62]: Too much `if` and indentation => put this heavy logic into a private function like `computeProductCover()`

File: views/templates/front/view.tpl [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+                            <form id="addtocart_{$product.id_product|intval}_{$product.id_product_attribute|intval}" action="{$link->getPageLink('cart')|escape:'html'}" method="post">
+                                <p class="hidden">
+                                    <input type="hidden" name="id_product" value="{$product.id_product|intval}" id="product_page_product_id"  />
+                                    <input type="hidden" name="add" value="1" />
+                                    <input type="hidden" name="token" value="{$token}" />
+                                    <input type="hidden" name="id_product_attribute" id="idCombination" value="{$product.id_product_attribute|intval}" />
+                                </p>
+                            </form>
+                        {/if}
+                        <a href="javascript:;" class="exclusive" onclick="WishlistBuyProduct('{$token|escape:'html':'UTF-8'}', '{$product.id_product}', '{$product.id_product_attribute}', '{$product.id_product}_{$product.id_product_attribute}', this, {$ajax});" title="{l s='Add to cart' mod='blockwishlist'}" rel="nofollow">{l s='Add to cart' mod='blockwishlist'}</a>
```
Comment (matks) [inline views/templates/front/view.tpl:70]: Line veeeeeery long 😄 your IDE did not complain ?

Comment (NeOMakinG) [reply views/templates/front/view.tpl:70]: Well anyway, this sound not be blocking the PR, it's a port from previous version, this file will be erased by my next work

File: classes/WishList.php (hunk 1 above)
Comment (Matt75) [reply classes/WishList.php:460]: This ObjectModel should be deprecated soon, he retrieved it from previous module version.
I guess new services will be created soon

File: .github/workflows/php.yml [hunk 9]
```diff
@@ -2,13 +2,11 @@ name: PHP tests
...
 jobs:
   php-linter:
-    name: PHP Syntax check 5.6|7.2|7.3
+    name: PHP Syntax check 7.2|7.3
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
         uses: actions/checkout@v2.0.0
-      - name: PHP syntax checker 5.6
-        uses: prestashop/github-action-php-lint/5.6@master
```
Comment (Quetzacoalt91) [inline .github/workflows/php.yml]: Module will be compatible with PHP 5.6, we should keep this step.

File: .github/workflows/php.yml [hunk 10]
```diff
@@ -26,22 +24,27 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        presta-versions: ['1.7.0.3', 'latest']
+        presta-versions: ['latest'] #todo Update it when 1.7.7.0 will be available
```
Comment (Quetzacoalt91) [inline .github/workflows/php.yml:27]: Compatibility will start with PS 1.7.6.0. The tag `prestashop/prestashop:1.7.6.0` has been pushed on Docker hub so we can use it here.

**Pull request merged**